deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml

```diff
@@ -340,7 +340,8 @@ data:
         data_stream:
           namespace: default
         streams:
-          - data_stream:
+          - id: container-log-${kubernetes.pod.name}-${kubernetes.container.id}
```

Comment:
So we need this ID in streams' level too? How is this different to the one that already exists at https://github.com/elastic/elastic-agent/blob/409682e5efba99f6b4ff034a503b1d0827a67b0b/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml#L333? 

cc: @gizas @MichaelKatsoulis

Reply:
Explained here: https://github.com/elastic/elastic-agent/pull/2573#discussion_r1181800166

The input level ID is used by the agent to keep track of the input state. For filestream specifically, each data stream becomes an independent instance of the filestream input in Beats. We need the ID at the stream level to give each filestream input in Beats a unique ID.

There is a one to many mapping between a `filestream` input in the agent policy and the number of filestream inputs created in Filebeat.

Reply:
Although, for kubernetes use case it seems like an one to one mapping, as each agent policy filestream input can only contain one stream, as they both point to one pod-container pair.

Reply:
The input itself uses variable substitution here, so the number of inputs would depend on the number of monitored containers wouldn't it?

Either way, the official examples should use a stream level ID as well so that someone modifying it will a correct example to start from.

Reply:
@cmacknz why we need both pod.name and container.id? Can we keep only container.id as it is unique afterall?

Reply:
I assume because a pod can contain multiple containers, this is the same thing we use for the input level ID.

Reply:
So reading https://github.com/elastic/elastic-agent/pull/2573#discussion_r1181800166 I understand that we need the `id` also on the `streams` level. Based on this the change looks good to me.

Reply:
Agreed, we don't have a place in the official docs where we document how to configure each input type in the standalone agent. I just pasted a summary of how this works in the reference YAML and the k8s manifests. Hopefully it helps clarify things.